spark_auto_mapper/automappers/with_column_base.py

```diff
@@ -62,23 +62,39 @@ def get_column_spec(self, source_df: Optional[DataFrame]) -> Column:
                 source_df=source_df, current_column=None, parent_columns=None
             )
             if self.skip_if_columns_null_or_empty:
+                column_type_dict = dict(source_df.dtypes)  # type: ignore
                 is_first_when_case = True
                 for column in self.skip_if_columns_null_or_empty:
                     column_to_check = f"b.{column}"
                     # wrap column spec in when
-                    column_spec = (
-                        when(
-                            col(column_to_check).isNull()
-                            | col(column_to_check).eqNullSafe(""),
-                            lit(None),
+                    if "array" in column_type_dict[column]:
+                        column_spec = (
+                            when(
+                                col(column_to_check).isNull()
+                                | (size(col(column_to_check)) == 0),
```

Comment:
see if we can use empty list for comparison

Reply:
eqNullSafe does not work for empty list. It just compares the value of a column